iiop-openjdk/src/main/java/org/wildfly/iiop/openjdk/IIOPSubsystemAdd.java

```diff
@@ -174,8 +171,9 @@ public void execute(DeploymentProcessorTarget processorTarget) {
 
         // if a security domain has been specified, add a dependency to the domain service.
         String securityDomain = props.getProperty(Constants.SECURITY_SECURITY_DOMAIN);
-        if (securityDomain != null)
-            builder.addDependency(SECURITY_DOMAIN_SERVICE_NAME.append(securityDomain));
+        if (securityDomain != null) {
+            builder.addDependency(context.getCapabilityServiceName(Capabilities.CAPABILITY_LEGACY_SECURITY_DOMAIN, securityDomain, null));
```

Comment:
If this is switching to build the service name using a capability, should the resource referencing the security domain also be defined as referencing a capability?

Reply:
Fixing.